src/PrettierEditProvider.ts

```diff
@@ -48,6 +49,10 @@ async function format(
     ) as any;
     const localPrettier = requireLocalPkg(fileName, 'prettier') as Prettier;
 
+    if (vscodeConfig.disableLanguages.includes(languageId)) {
+        return text;
+    }
```

Comment:
Instead of a noop we should just not register the formatter for those languages. To be able to use an other formatter for instance.

in extension.ts:
`languageSelector - disableLanguages`
`rangeLanguageSelector - disableLanguages`

This means "restart required" :smile:

Reply:
Ok, never mind, forgot multi-root......

Reply:
In fact, It could be a `resource: "window"` scope settings. Like our old `*Enabled` settings.

Reply:
IMO `"resource": "window"` should not exist because it doesn't do anything in multi-root. Seems annoying that there's no way to skipping formatting on a per-resource level.

Maybe a feature request to `return null` from a formatter to skip it in VSCode?

Reply:
https://github.com/Microsoft/vscode/issues/38945

Reply:
On VSCode's side might be a good thing. :+1:

Reply:
It actually seems to work if you throw an error... https://github.com/Microsoft/vscode/issues/38945#issuecomment-346319330

```ts
    if (vscodeConfig.disableLanguages.includes(languageId)) {
        throw new Error('Skip me!');
    }
```

Reply:
Seems great!  just waiting for a confirmation on that *functionality*. 
API says:
>Returns: A set of text edits or a thenable that resolves to such. The lack of a result can be signaled by returning undefined, null, or an empty array.